app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -794,39 +476,589 @@ public void onCharacteristicChanged(BluetoothGatt gatt, BluetoothGattCharacteris
             final PowerManager.WakeLock wakeLock1 = JoH.getWakeLock("DexCollectionService", 60000);
             try {
                 final byte[] data = characteristic.getValue();
-                final String hexdump = HexDump.dumpHexString(data);
-                if (!hexdump.contains("0x00000000 00      ")) {
-                    static_last_hexdump = hexdump;
-                }
-                Log.i(TAG, "onCharacteristicChanged entered " + hexdump);
+
                 if (data != null && data.length > 0) {
                     setSerialDataToTransmitterRawData(data, data.length);
+
+                    final String hexdump = HexDump.dumpHexString(data);
+                    //if (!hexdump.contains("0x00000000 00      ")) {
+                    if (data.length > 1 || data[0] != 0x00) {
```

Comment:
Here optimized to not search the string but look at the bytes for the same effect